### PR TITLE
Dimitrisstaratzis/arrow updates

### PR DESCRIPTION
**Pull request by DimitrisStaratzis**

### CODE REVIEW

File: src/main/java/io/tiledb/java/api/Query.java [hunk 1]
```diff
@@ -1304,6 +1304,47 @@ public Object getCoordinates() throws TileDBError {
...
+   */
+  public HashMap<String, Pair<Long, Long>> resultBufferElementsNIO(int typeSize)
+      throws TileDBError {
+    HashMap<String, Pair<Long, Long>> result = new HashMap<String, Pair<Long, Long>>();
+    for (Map.Entry<String, Pair<ByteBuffer, ByteBuffer>> entry : byteBuffers_.entrySet()) {
+      String name = entry.getKey();
+
+      // Fixed-sized
+      if (entry.getValue().getFirst() == null) {
+        BigInteger val_nbytes = buffer_sizes_.get(name).getSecond().getitem(0);
```
Comment (ihnorton) [inline src/main/java/io/tiledb/java/api/Query.java:1324]: Should we cast this to a known max size integer so that we get an error if it is too big?

**Review by ihnorton — CHANGES_REQUESTED** [review]
One question, and do you mind to add a small test before merging?

File: src/main/java/io/tiledb/java/api/Query.java (hunk 1 above)
Comment (DimitrisStaratzis) [reply src/main/java/io/tiledb/java/api/Query.java:1324]: BigInteger will throw an ArithmeticException when the result is out of the supported range. I am not sure I understood your comment, sorry..

Comment (DimitrisStaratzis) [reply src/main/java/io/tiledb/java/api/Query.java]: Added the tests and made some minor changes to the code.

**Review by ihnorton — APPROVED** [review]

**Pull request merged**